### PR TITLE
object-manipulation-2 challenge - Ken Kieu

### PR DESCRIPTION
**Pull request by kenkieu**
Completed object-manipulation-2 challenge.
image

### CODE REVIEW

File: object-manipulation-2/src/pick.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// look over each item in source input
+// if the key inputs value is the same as the keys inputs, and the value of the source is not undefined
+// set the property and value of the source input to new obj
+// return new obj
+
+function pick(source, keys) {
+  var outputObj = {};
+  for (var i = 0; i < keys.length; i++) {
+    for (var key in source) {
+      if (key === keys[i] && source[key] !== undefined) {
```
Comment (codingcodymiller) [inline object-manipulation-2/src/pick.js:16]: What happens if you remove this `for...in` loop and only keep the second expression in your condition here?

Comment (codingcodymiller) [reply object-manipulation-2/src/pick.js:16]: Meaning `source[keys[i]] !== undefined`

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Great work!! A couple notes for you:

File: object-manipulation-2/src/pick.js (hunk 1 above)
Comment (kenkieu) [reply object-manipulation-2/src/pick.js]: Oh, that makes a lot of sense. No need to loop through when we're just comparing to source for the specific keys. Revised, thanks!

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**